Framework/Geometry/src/Instrument/Component.cpp

```diff
@@ -393,6 +387,152 @@ Quat Component::getRotation() const {
       return m_parent->getRotation() * m_rot;
   }
 }
+*/
+
+/// Helper for legacy access mode. Returns the index of the component.
+size_t Component::index() const {
+  return m_map->componentIndex(this->getComponentID());
+}
+
+bool Component::hasComponentInfo() const {
+  const IComponent *root = m_base;
+  while (auto parent = root->getBareParent())
+    root = parent;
+  auto instrument = dynamic_cast<const Instrument *>(root);
+  return m_map->hasComponentInfo(instrument);
+}
+
+/// Return the relative position to the parent Component
+Kernel::V3D Component::getRelativePos() const {
+  if (m_map) {
+
+    if (hasComponentInfo() && !m_map->componentInfo().isDetector(index())) {
```

Comment:
Because this method will get called during the creation of `DetectorInfo`. i.e. `Detector::getPos()` But at that point the `DetectorInfo` won't exist and can't be used by the `ComponentInfo` so we have to do something different for `Detectors`. This is part of the clean-up that will be needed. i.e. making the `InfoComponentVisitor` create both `ComponentInfo` and `DetectorInfo` in one hit.